test/sequential/test-child-process-execsync.js

```diff
@@ -12,6 +12,13 @@ var start = Date.now();
 var err;
 var caught = false;
 
+// Verify that stderr is not accessed when a bad shell is used
+(function() {
+  assert.throws(function() {
+    execSync('exit -1', {shell: 'no_shell'});
+  }, /spawnSync no_shell ENOENT/);
+})();
+
```

Comment:
There's a similar test further down below.  This new test should probably be added there.